addons/account/models/account_move.py

```diff
@@ -3750,6 +3750,17 @@ def invalidate_cache(self, fnames=None, ids=None):
                 self.env['account.move'].invalidate_cache(ids=move_ids)
         return super().invalidate_cache(fnames=fnames, ids=ids)
 
+    @api.model
+    def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_uid=None):
+        if operator == 'ilike':
+            args = ['|', '|',
+                    ('name', '=', name),
+                    ('move_id.name', '=', name),
+                    ('product_id.display_name', '=', name)]
```

Comment:
Are we sure about the product? Isn't it copied in the `name` anyway?

Reply:
I just used the code from `name_get`

Reply:
Then I think `('product_id', 'ilike', name)` would be better